internal/worker/dbaccessor/worker_integration_test.go

```diff
@@ -128,6 +130,16 @@ func (s *integrationSuite) TearDownTest(c *gc.C) {
 	if dqlite.Enabled && s.worker != nil {
 		workertest.CleanKill(c, s.worker)
 	}
+	if s.db != nil {
+		s.db.Close()
+	}
+
+	// We can't for what ever reason call s.DqliteSuite.TearDownTest here
```

Comment:
This appears to be a concerning statement which I am not sure we should be happy with. Looking at this suite, `integrationSuite` it embeds `dqliteAppIntegrationSuite` which in turn embeds `DqliteSuite`.  And as you say calling that `DqliteSuite` test tear down is deliberately avoided. 
But having this black magic leak out into our test suites is a big problem for me, especially since we now need to sprinkle magic dust like the code below all around the place.
Can we instead address the root cause?

Reply:
> Can we instead address the root cause?

Maybe... I'll have to see. It maybe a bug in dqlite

Reply:
Right, I spent some time investigating. Turns out we can never call `Close` twice on a dqlite app. I'll speak with @manadart to see what we can do about it.